query.go

```diff
@@ -80,6 +80,18 @@ func GetNamedVideoSource(
 	return newVideoSourceFromDriver(d, selectedMedia, logger)
 }
 
+func GetStatusVideoSource(
```

Comment:
How do you intend on using this? I thought we wanted to just always requery?

Reply:
Got it. It seems better to expose a way to access the driver that you have from the media source instead. The underlying unexported type has the driver so you can maybe make a method to get the driver/status from it instead of doing constraint evaluation again which can be a little expensive

Reply:
So we'd have to query for a new driver each time we check the status. This is because the status on any given driver is static. So if we saved the driver in a variable and kept checking its status we'd always get "running" back even after it's unplugged. To avoid running the SelectSettings algorithm over and over I added a method `SelectDriverByDeviceId` to filter based solely on the randomly generated UUID of the device driver grabbed from `GetStatusVideoSource` (renamed `GetDriverVideoSource`). We'd use the two methods like so
```
	if attrs.Path != "" {
		cam, err := tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
		if err != nil {
			return cam, err
		}

		goutils.PanicCapturingGo(func() {
			ticker := time.NewTicker(500 * time.Millisecond)
			defer ticker.Stop()
			defer goutils.TryClose(ctx, cam)

			name := resolveSourceName(attrs.Path, false)
			d, err := gostream.GetDriverVideoSource(name, constraints, logger)
			if err != nil {
				return
			}
			uuid := d.ID()
			for {
				select {
				case <-ticker.C:
					if d := gostream.SelectDriverByDeviceId(uuid); d == nil {
						return
					}
				case <-ctx.Done():
					return
				}
			}
		})

		return cam, err
	}

```

Reply:
I did not add any documentation yet just in case we need more changes to function signatures.